test/software/amazon/ion/impl/lite/BaseIonSequenceLiteTestCase.java

```diff
@@ -0,0 +1,655 @@
+    public void sublistRemoveIndexConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.remove(0);
+    }
+
+    @Test
+    public void sublistRemoveObject() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final IonInt ionValue = (IonInt) parent.get(2);
+        assertTrue(sublist.remove(ionValue));
+        assertEquals(2, sublist.size());
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistRemoveObjectConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.remove(parent.get(2));
+    }
+
+    @Test
+    public void sublistRemoveAll() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final List<IonValue> toRemove = Arrays.asList(parent.get(2), parent.get(3));
+        assertTrue(sublist.removeAll(toRemove));
+        assertEquals(1, sublist.size());
+        for (IonValue v : toRemove) {
+            assertFalse(sublist.contains(v));
+        }
+    }
+
+    @Test
+    public void sublistRemoveAllNotInList() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        final List<IonValue> toRemove = Arrays.asList(parent.get(0), parent.get(1));
+        assertFalse(sublist.removeAll(toRemove));
+        assertEquals(3, sublist.size());
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistRemoveAllConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.removeAll(Collections.singletonList(parent.get(2)));
+    }
+
+    @Test
+    public void sublistIndexOf() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+
+        assertEquals(0, sublist.indexOf(parent.get(2)));
+        assertEquals(-1, sublist.indexOf(parent.get(0)));
+        assertEquals(-1, sublist.indexOf(SYSTEM.newInt(99)));
+    }
+
+    @Test(expected = ConcurrentModificationException.class)
+    public void sublistIndexOfConcurrentModification() {
+        final IonList parent = SYSTEM.newList(new int[]{0, 1, 2, 3, 4, 5, 6});
+        final List<IonValue> sublist = parent.subList(2, 5);
+        parent.remove(0);
+
+        sublist.indexOf(parent.get(2));
+    }
+
+    @Test
+    public void sublistLastIndexOf() {
```

Comment:
not confident in this test;  how do we know it's returning the last index (instead of the first)?

should probably say the same for any `indexOf()` tests

Reply:
`IonSequence` uses reference equality and does not allow for duplicates so `indexOf` and `lastIndexOf` are equivalent. 

This is somewhat explained on `IonSequence#lastIndexOf` javadoc, but I'll make it more explicit